RecoHGCal/TICL/plugins/TrackstersMergeProducer.cc

```diff
@@ -129,16 +129,14 @@ TrackstersMergeProducer::TrackstersMergeProducer(const edm::ParameterSet &ps, co
   produces<std::vector<TICLCandidate>>();
 }
 
-void TrackstersMergeProducer::fillTile(TICLTracksterTiles &tracksterTile,
-                                       const std::vector<Trackster> &tracksters,
-                                       TracksterIterIndex tracksterIteration) {
+void TrackstersMergeProducer::fillTile(TICLTracksterTiles &tracksterTile, const std::vector<Trackster> &tracksters) {
   int tracksterId = 0;
   for (auto const &t : tracksters) {
-    tracksterTile.fill(tracksterIteration, t.barycenter().eta(), t.barycenter().phi(), tracksterId);
+    tracksterTile.fill(t.ticlIteration(), t.barycenter().eta(), t.barycenter().phi(), tracksterId);
     LogDebug("TrackstersMergeProducer") << "Adding tracksterId: " << tracksterId << " into bin [eta,phi]: [ "
-                                        << tracksterTile[tracksterIteration].etaBin(t.barycenter().eta()) << ", "
-                                        << tracksterTile[tracksterIteration].phiBin(t.barycenter().phi())
-                                        << "] for iteration: " << tracksterIteration << std::endl;
+                                        << tracksterTile[t.ticlIteration()].etaBin(t.barycenter().eta()) << ", "
+                                        << tracksterTile[t.ticlIteration()].phiBin(t.barycenter().phi())
+                                        << "] for iteration: " << t.ticlIteration() << std::endl;
 
```

Comment:
`t.ticleIteration()` will return the very same value for all tracksters in this method, since they all belong to the same iteration. Would it make sense to revert to the previous logic, leave the signature as it was (i.e. with the last `tracksterIteration`) to avoid unnecessary calls?

Reply:
OK.